src/core.js

```diff
@@ -1288,10 +1288,20 @@ define([
                 _this.setUnsavedDuplicateNodeId(false);
             } else if (e.property === 'defaultValue' && e.mug.__className === 'Item') {
                 _this.setUnsavedDuplicateChoiceValue(false);
+            } else if (e.property === 'dataParent') {
+                // TODO Dirty write hack that needs to be replace
+                e.mug.options.isDataOnly = true;
+                _this.data.core.form.insertMug(e.mug.parentMug, e.mug, 'into');
```

Comment:
Insert the parent as a child of it's child? That doesn't seem right. Seems like we need a new function on the form that knows how to change the data path of a mug: `e.mug.form.updateDataParent(e.mug)`.